storage-proofs/core/src/parameter_cache.rs

```diff
@@ -305,12 +313,14 @@ where
 
         // generate (or load) verifying key
         let cache_path = ensure_ancestor_dirs_exist(parameter_cache_verifying_key_path(&id))?;
-        read_cached_verifying_key(&cache_path)
-            .or_else(|_| write_cached_verifying_key(&cache_path, generate()?))
+        match read_cached_verifying_key(&cache_path) {
```

Comment:
I guess `or_else()` could be kept here? Any reasons to change, other than a preference for pattern matching?

Reply:
the types got really messy, because of two different Error types, and it got really hard to read